src/decompiler.cc

```diff
@@ -208,33 +218,27 @@ struct Decompiler : ModuleContext {
         return;
       }
       case ExprType::LocalGet: {
-        auto& lg = *cast<LocalGetExpr>(&e);
-        ss << lg.var.name();
-        PushSStream();
+        Get(*cast<LocalGetExpr>(&e));
         return;
       }
       case ExprType::GlobalGet: {
-        auto& gg = *cast<GlobalGetExpr>(&e);
-        ss << gg.var.name();
-        PushSStream();
+        Get(*cast<GlobalGetExpr>(&e));
         return;
       }
       case ExprType::LocalSet: {
-        auto& ls = *cast<LocalSetExpr>(&e);
-        WrapChild(stack.back(), ls.var.name() + " = ", "");
-        stack.back().needs_bracketing = true;
+        Set(*cast<LocalSetExpr>(&e));
         return;
       }
       case ExprType::GlobalSet: {
-        auto& gs = *cast<GlobalSetExpr>(&e);
-        WrapChild(stack.back(), gs.var.name() + " = ", "");
-        stack.back().needs_bracketing = true;
+        Set(*cast<GlobalSetExpr>(&e));
         return;
       }
       case ExprType::LocalTee: {
         auto& lt = *cast<LocalTeeExpr>(&e);
-        WrapChild(stack.back(), lt.var.name() + " = ", "");
-        stack.back().needs_bracketing = true;
+        Set(lt);
+        if (stack_depth == 1) {  // Tee is the only thing on there.
```

Comment:
Otherwise what happens? If `stack_depth > 1`, the value is still on the stack? We don't need to re-get here?

So if given
```
A:
i32.const 1
local.tee 0
i32.const 2
i32.add

B:
i32.const 1
i32.const 2
local.tee 0
i32.add
```

I would expect to see:
```
A:
a = 1
a + 2

B:
1 + (a = 2)
```

oh ok I answered my own question. The set still goes on the stack and gets parenthesized, so it just looks like an expression in that case. Nice!

Reply:
Will add a comment to clarify what happens in the `else` case :)